src/wasm-interpreter.h

```diff
@@ -1418,7 +1428,26 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
     WASM_UNREACHABLE("throw");
   }
   Flow visitRethrow(Rethrow* curr) { WASM_UNREACHABLE("unimp"); }
-  Flow visitThrowRef(ThrowRef* curr) { WASM_UNREACHABLE("unimp"); }
+  Flow visitThrowRef(ThrowRef* curr) {
+    NOTE_ENTER("ThrowRef");
+    Flow flow = visit(curr->exnref);
+    if (flow.breaking()) {
+      return flow;
+    }
+    const auto& exnref = flow.getSingleValue();
+    NOTE_EVAL1(exnref);
+    if (exnref.isNull()) {
+      trap("null ref");
+    }
+    const auto& exnData = exnref.getExnData();
+    WasmException exn;
+    exn.tag = exnData->tag;
+    for (auto item : exnData->payload) {
+      exn.values.push_back(item);
+    }
+    throwException(exn);
```

Comment:
```suggestion
    throwException(exnData);
```
This can simplified if we change `WasmException` to take `ExnData` literal. See my comment on line 49-53.